File: client/src/app/models/bill.ts
import { BillType } from "./bill-type";

export interface Bill {
    id: number;
    value: number;
    month: number;
    year: number;
    billType: BillType;
}

export interface NewBillDto {
    value: number;
    month: number;
    year: number;
    typeId: number;
    userId: number;
}

File: Paybills.API/Entities/Bill.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Paybills.API.Entities
{
    [Table("Bills")]
    public class Bill
    {
        public int Id { get; set; }
        public BillType BillType { get; set; }
        public float Value { get; set; }
        public int Month { get; set; }
        public int Year { get; set; }
        public DateTime? DueDate { get; set; }
        public bool Paid { get; set; }
        public ICollection<AppUser> Users { get; set; }
    }
}